posts/change-data-capture-cloudsql-postgres-with-apache-airflow-1.md

```diff
@@ -0,0 +1,1016 @@
+
+To create this pool is simple. Just navigate in the top bar to `Admin -> Pools`, then hit the `+` sign to add a new record. Choose a pool name and description to your liking, but choose the number of slots as `1`.
+
+![Pools](../assets/cdc-cloudsql-1/cdc_cloudsql_airflow_2021-01-14_at_11.05.02_AM.png)
+
+Will add the name of the pool to your task definition below and that's it.
+
+## Step 7: Build Your Pipeline in Airflow
+
+  > *If you are familiar with the foundational concepts of Airflow, you can skip to the next subsection for the `DAG Workflow`.*
+
+Airflow pipeline design rests on the foundational structures of `hooks` and `operators`. Hooks facilitate connections and repository-type actions, whereas operators use hooks in combinations with domain-specific logic and error handling in order to accomplish a task. Airflow comes pre-packaged with many of these hooks and operators, all built to work with services we use regularly. For example, `BashOperator` allows you to run a bash command directly in-line. A simple task using a this operator might look like:
+
+```python
+hello_world = BashOperator(
+    task_id='get_schema',
+    bash_command='echo "hello, world"'
+)
+```
+
+Getting more advanced, a `PostgresOperator` executes SQL code in a specific PostgreSQL database using a `PostgresHook` to establish the connection, while templating your command and implementing a `run` function to execute it. A task involving this operator might look like:
+
+```python
+insert_row = PostgresOperator(
+    task_id='insert_row',
+	  postgres_conn_id='default_conn_id',
+    sql='INSERT INTO table_1 VALUES (%s, %s, %s)',
+    parameters=(1, 'a very important string', datetime.now())
+)
+```
+
+### DAG Workflow
+
+Because of the great flexibility of Airflow, you can also use hooks to create custom operators, just as well as you can create custom hooks. In this post, we will create a custom `CloudSqlCsvExportOperator` using an existing `CloudSqlHook`, a hook designed to facilitate several types of operations on CloudSQL instances. First, we will use the `BashOperator` to simulate our workflow and check dependencies.
+
+In the DAG workflow, it is extremely advantageous to also create generic tasks that serve only as checkpoints in the workflow. This facilitates best practices like idempotency, task grouping, workflow readability, and traceability. This is accomplished by a stand-in operator called the `DummyOperator`. In the same effort, the `DummyOperator` is an excellent way to first build your workflow and visualize task dependencies. Combining this with a `BashOperator` allows us to add a "Hello, world" level of functionality to this. This is what we will do first.
```

Comment:
```suggestion
In the DAG workflow, it is extremely advantageous to also create generic tasks that serve only as checkpoints in the workflow. This facilitates best practices like idempotency, task grouping, workflow readability, and traceability. This is accomplished by a stand-in operator called the `DummyOperator`. Additionally, the `DummyOperator` is an excellent way to first build your workflow and visualize task dependencies. Combining this with a `BashOperator` allows us to add a "Hello, world" level of functionality to this. To implement the `DummyOperator`:
```

Reply:
@jwitz I totally re-wrote this. Let me know what you think.

In the DAG workflow, it is extremely advantageous to also create generic tasks that serve only as checkpoints in the workflow. This facilitates best practices like idempotency, task grouping, workflow readability, and traceability. This is accomplished by a stand-in operator called the `DummyOperator`. Building your whole DAG first with all `DummyOperator` tasks is an excellent way to visualize task dependencies and control the flow of tasks. As a next step, combining this with a `BashOperator` allows us to add a "Hello, world" level of functionality to this and test the flow of the DAG itself. To do this:

Reply:
Using some more metaphorical language to communicate your point

"Outlining your DAG using DummyOperator tasks is an excellent way to visualize task dependencies and control the flow of tasks. As a next step, we can use a BashOperator to add a "Hello, world" level of functionality and test the flow of the DAG. To do this:"

From what I understand, we're essentially building an outline. I think it's important to get that word in there as a cornerstone for what you're trying to communicate.